src/features/metadata-instance-editor/__tests__/Instance.test.js

```diff
@@ -147,82 +148,149 @@ const userDefinedTemplateField = [
-        );
-        expect(wrapper).toMatchSnapshot();
-    });
+    test.each`
+        isFolderInstance
+        ${true}
+        ${false}
+    `(
+        'should correctly render templated metadata instance when isFolderInstance is $isFolderInstance',
+        ({ isFolderInstance }) => {
+            render(
+                <Instance
+                    cascadePolicy={{ canEdit: true, id: 'hello' }}
+                    data={data}
+                    dataValue="value"
+                    intl={intl}
+                    isCascadingPolicyApplicable
+                    isFolderInstance={isFolderInstance}
+                    shouldShowCascadingOptions
+                    onModification={jest.fn()}
+                    onRemove={jest.fn()}
+                    onSave={jest.fn()}
+                    canEdit
+                    template={{
+                        fields,
+                    }}
+                />,
+            );
+
+            const editButton = screen.queryByLabelText('Edit Metadata');
+            if (editButton) {
+                fireEvent.click(editButton);
+            }
+            const metadataInstanceEditor = screen.queryByTestId('metadata-cascade-policy');
+            if (isFolderInstance) {
+                expect(metadataInstanceEditor).toBeInTheDocument();
+            } else {
+                expect(metadataInstanceEditor).not.toBeInTheDocument();
+            }
+        },
+    );
+
+    test.each`
+        cascadePolicy                     | isCascadingPolicyApplicable
+        ${{ canEdit: true, id: 'hello' }} | ${true}
+        ${{ canEdit: true, id: 'hello' }} | ${false}
+        ${{ canEdit: true }}              | ${true}
+        ${{ canEdit: true }}              | ${false}
+        ${null}                           | ${true}
+        ${null}                           | ${false}
+    `(
+        'should correctly render templated metadata instance when cascadePolicy is $cascadePolicy and isCascadingPolicyApplicable is $isCascadingPolicyApplicable',
+        ({ cascadePolicy, isCascadingPolicyApplicable }) => {
+            const { container } = render(
+                <Instance
+                    cascadePolicy={cascadePolicy}
+                    data={data}
+                    dataValue="value"
+                    intl={intl}
+                    isCascadingPolicyApplicable={isCascadingPolicyApplicable}
+                    isFolderInstance
+                    shouldShowCascadingOptions
+                    onModification={jest.fn()}
+                    onRemove={jest.fn()}
+                    onSave={jest.fn()}
+                    canEdit
+                    template={{
+                        fields,
+                    }}
+                />,
+            );
+
+            const editButton = screen.queryByLabelText('Edit Metadata');
+            if (editButton) {
+                fireEvent.click(editButton);
+            }
+            expect(container).toMatchSnapshot();
```

Comment:
could we switch to schecking if the component is in `TemplateInstance` with `toBeInTheDocument`?

I don't think the snapshots are helping in this case since the file is so large it's hard to verify

Reply:
Updated this to check when the instance is a template and is editing, that the correct toggle is showing

Reply:
We check that an instance is template or custom in the test above this one